### PR TITLE
Support Interactions

### LINKED ISSUE #1100: Discord slash commands
Discord is deploying slash commands.

https://github.com/discord/discord-api-docs/pull/2295/commits/30e6cca446940ca9f51a994f3a9cb63917f7751b#diff-07978ef9d619062fb1afbb70ea065124531246743444f4354cc9e6c8879f8780

### PR DESCRIPTION
**Pull request by JustCat80**
Recreated from #1210 for hacktober

## Notes
https://discord.com/developers/docs/interactions/receiving-and-responding#interactions
https://discord.com/developers/docs/interactions/application-commands

Auto Updating Docs (built by @DonovanDMC)
https://eris.owo-whats-this.dev/interactions/

Adds support for Interactions, which adds the ability to reply to [Application Commands](https://discord.com/developers/docs/interactions/application-commands), [Message Components](https://discord.com/developers/docs/interactions/message-components), and other related things.
Also adds support for [Application Commands](https://discord.com/developers/docs/interactions/application-commands)
~~For Message Components, check out bsian's pr #1189~~

Please, let me know of any Criticisms you may have and/or changes that should be made.
### Status of being merged:
~~_As per [Abal's comment on slash commands](https://github.com/abalabahaha/eris/issues/1100#issuecomment-812851950): 
"Eris doesn't plan to implement full support for slash commands until the API gets improved+stabilized."_~~

**As per [this discord post](https://support-dev.discord.com/hc/en-us/articles/4404772028055) this is now required soon because as of april 2022 message content will become a privileged intent, and will likely be merged when discord slows down the rate of changes**
## TODO:
Fix https://github.com/abalabahaha/eris/pull/1210#issuecomment-933005417
Add WebhookClient
## Additions:
_updated as of commit 9f1d485 and may not reflect exact changes of commits after it_
### Events:
- Client Event: interactionCreate
#### Structures:
- Interaction
- PingInteraction
- CommandInteraction
- ComponentInteraction
- AutocompleteInteraction
- UnknownInteraction
#### PingInteraction Methods:
- acknowledge
- pong
#### CommandInteraction Methods:
- acknowledge
- createFollowup
- createMessage
- defer
- deleteMessage
- deleteOriginalMessage
- editMessage
- editOriginalMessage
- getOriginalMessage
#### ComponentInteraction Methods:
- acknowledge
- createFollowup
- createMessage
- defer
- deferUpdate
- deleteMessage
- deleteOriginalMessage
- editMessage
- editOriginalMessage
- editParent
- getOriginalMessage
#### AutocompleteInteraction Methods:
- acknowledge
- result
#### UnknownInteraction Methods:
UnknownInteraction has all the methods of the Interaction types listed above, just in case you know what will work
#### Client Methods:
- createInteractionResponse
- bulkEditCommandPermissions
- bulkEditCommands
- bulkEditGuildCommands
- createCommand
- createGuildCommand
- deleteCommand
- deleteGuildCommand
- editCommand
- editCommandPermissions
- editGuildCommand
- getCommand
- getCommandPermissions
- getCommands
- getGuildCommand
- getGuildCommandPermissions
- getGuildCommands
#### Guild Methods:
- bulkEditCommandPermissions
- bulkEditCommands
- createCommand
- deleteCommand
- editCommand
- editCommandPermissions
- getCommand
- getCommandPermissions
- getCommands
- getGuildCommandPermissions
#### Constants:
- new MessageFlags value
- InteractionTypes
- InteractionResponseTypes
- CommandOptionTypes
- CommandPermissionTypes
- CommandTypes
- ButtonStyles
#### Misc:
- [Endpoints](https://github.com/abalabahaha/eris/blob/1833fcf/lib/rest/Endpoints.js)
- ts docs

### CODE REVIEW

**Comment by JustCat80** [comment]
Fixed https://github.com/abalabahaha/eris/pull/1210#issuecomment-933005417 in 2292ded but I'm not sure its the best way to do it, lmk what yall think

File: index.d.ts [hunk 1]
```diff
@@ -1709,18 +1903,23 @@ declare namespace Eris {
...
     ): Promise<Webhook>;
+    createCommand(command: ApplicationCommandStructure): Promise<ApplicationCommand>;
     createGroupChannel(userIDs: string[]): Promise<GroupChannel>;
     createGuild(name: string, options?: CreateGuildOptions): Promise<Guild>;
+    createGuildCommand(guildID: string, command: ApplicationCommandStructure): Promise<ApplicationCommand>;
     createGuildEmoji(guildID: string, options: EmojiOptions, reason?: string): Promise<Emoji>;
     createGuildFromTemplate(code: string, name: string, icon?: string): Promise<Guild>;
     createGuildSticker(guildID: string, options: CreateStickerOptions, reason?: string): Promise<Sticker>;
     createGuildTemplate(guildID: string, name: string, description?: string | null): Promise<GuildTemplate>;
+    createInteractionResponse(interactionID: string, interactionToken: string, options: InteractionOptions, file: FileContent | FileContent[]): Promise<void>;
```
Comment (DonovanDMC) [inline index.d.ts]: File is marked as required
```suggestion
    createInteractionResponse(interactionID: string, interactionToken: string, options: InteractionOptions, file?: FileContent | FileContent[]): Promise<void>;
```

File: lib/Constants.js [hunk 2]
```diff
@@ -208,7 +208,8 @@ module.exports.MessageFlags = {
     IS_CROSSPOST:           1 << 1,
     SUPPRESS_EMBEDS:        1 << 2,
     SOURCE_MESSAGE_DELETED: 1 << 3,
-    URGENT:                 1 << 4
+    URGENT:                 1 << 4,
+    EPHEMERAL:              1 << 6
```
Comment (Awoocado) [inline lib/Constants.js]:
```suggestion
    EPHEMERAL:              1 << 6,
    LOADING:                1 << 7
```

File: index.d.ts [hunk 3]
```diff
@@ -1281,6 +1425,7 @@ declare namespace Eris {
       SUPPRESS_EMBEDS: 4;
       SOURCE_MESSAGE_DELETED: 8;
       URGENT: 16;
+      EPHEMERAL: 64;
```
Comment (Awoocado) [inline index.d.ts:1428]:
```suggestion
      EPHEMERAL: 64;
      LOADING: 128;
```

**Review by bsian03 — APPROVED** [review]
yolo

**Pull request merged**